ompi/mca/osc/ucx/osc_ucx_component.c

```diff
@@ -741,8 +754,8 @@ static int component_select(struct ompi_win_t *win, void **base, size_t size, in
     if (flavor == MPI_WIN_FLAVOR_ALLOCATE || flavor == MPI_WIN_FLAVOR_CREATE ||
             flavor == MPI_WIN_FLAVOR_SHARED) {
         memcpy(my_info, base, sizeof(uint64_t));
-    } else {
-        memcpy(my_info, &zero, sizeof(uint64_t));
+    } else if (flavor == MPI_WIN_FLAVOR_DYNAMIC) {
+        memcpy(my_info, &dynamic_base, sizeof(uint64_t));
```

Comment:
Why do we use `memcpy` here? And why do you copy the pointer to `dynamic_base`, not `dynamic_base` itself?

Reply:
because we want to copy the address itself, not the content of the memory that it is pointing to. Here we are encapsulating each process's data and state memory addresses.

Reply:
I think in that case the `memcpy` is a) superfluous and b) incorrect on 32-bit platforms. Why not 
```suggestion
        *my_info = dynamic_base;
```
and similar above:

```C
*my_info = base;
```

That leads me to another question: where is `my_info` free'd and I why isn't it just stack-based `uint64_t[2]`?

Reply:
I think uint64_t[2] should suffice here. I do not see my_info getting freed anywhere.